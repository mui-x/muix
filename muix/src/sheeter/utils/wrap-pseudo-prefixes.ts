// ({}, [':hover', ':active']) => {':hover': {':active': {}}}
export const wrapPseudoPrefixes = (rules: {}, pseudoPrefixes: string[]) => {
  if (pseudoPrefixes.length === 0) return rules
  let res = rules
  for (let i = pseudoPrefixes.length - 1; i >= 0; i--)
      res = { [pseudoPrefixes[i]]: res }
  return res
}