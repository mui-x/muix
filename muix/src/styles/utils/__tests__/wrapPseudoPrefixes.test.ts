import { wrapPseudoPrefixes } from "../wrap-pseudo-prefixes";

test("wrapPseudoPrefixes", () => {
  const res = wrapPseudoPrefixes({ color: "red" }, [
    ":hover",
    "@media (min-width: 768px)"
  ]);
  expect(res).toMatchSnapshot();
});
