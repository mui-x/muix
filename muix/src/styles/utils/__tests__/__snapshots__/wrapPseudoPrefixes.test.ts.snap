// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapPseudoPrefixes 1`] = `
Object {
  ":hover": Object {
    "@media (min-width: 768px)": Object {
      "color": "red",
    },
  },
}
`;
